src/frontend/src/optimizer/plan_node/stream_dynamic_filter.rs

```diff
@@ -33,18 +36,41 @@ pub struct StreamDynamicFilter {
     pub base: PlanBase<Stream>,
     core: generic::DynamicFilter<PlanRef>,
     cleaned_by_watermark: bool,
+    condition_always_relax: bool,
 }
 
 impl StreamDynamicFilter {
     pub fn new(core: DynamicFilter<PlanRef>) -> Self {
         let watermark_columns = core.watermark_columns(core.right().watermark_columns()[0]);
 
-        // TODO: derive from input
+        // TODO(st1page): the condition is wrong, introduce monotonically increasing property of the node
+        // TODO(st1page): https://github.com/risingwavelabs/risingwave/pull/13984
```

Comment:
```suggestion
        // TODO(st1page): here we just check if RHS
        // is a `StreamNow`. It will be generalized to more cases
        // by introducing monotonically increasing property of the node in https://github.com/risingwavelabs/risingwave/pull/13984.
```